CMakeLists.txt

```diff
@@ -4,16 +4,47 @@
 # cmake-format: on
 
 cmake_minimum_required(VERSION 3.23)
+
 project(
-        beman.inplace_vector
-        VERSION 1.0.0
-        DESCRIPTION "A dynamically-resizable vector with fixed capacity and embedded storage"
-        LANGUAGES CXX
-)
+  beman.inplace_vector
+  VERSION 1.0.0
+  DESCRIPTION
+    "A dynamically-resizable vector with fixed capacity and embedded storage"
+  LANGUAGES CXX)
+
+add_library(beman.inplace_vector INTERFACE)
+target_include_directories(
+  beman.inplace_vector
+  INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+            $<INSTALL_INTERFACE:include>)
+
+# Add the current source directory to the Include directories
+include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 
-include(CTest)
-add_subdirectory(src/beman/inplace_vector)
+# Set the target Include directories for the InplaceVector library
+target_include_directories(
+  beman.inplace_vector
+  INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
+
+# Install the InplaceVector library to the appropriate destination
+install(
+  TARGETS beman.inplace_vector
+  EXPORT ${TARGETS_EXPORT_NAME}
+  DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+# Convert the project name to lower case
+string(TOLOWER ${CMAKE_PROJECT_NAME} CMAKE_LOWER_PROJECT_NAME)
+
+# Install the header files to the appropriate destination
+install(
+  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${CMAKE_LOWER_PROJECT_NAME}
+  FILES_MATCHING
+  PATTERN
+    "${CMAKE_CURRENT_SOURCE_DIR}/include/beman/inplace_vector/inplace_vector.hpp"
+)
```

Comment:
Some tightening:

```suggestion
# Install the header files to the appropriate destination
install(
  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/
  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${CMAKE_PROJECT_NAME}
  FILES_MATCHING
  PATTERN
    "${CMAKE_CURRENT_SOURCE_DIR}/include/beman/inplace_vector/inplace_vector.hpp"
)
```

I think this could be improved with the `FILE_SET` approach, but that can come in a later PR.